azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java

```diff
@@ -1074,6 +1077,11 @@ private boolean isServiceRequired() {
    * @throws ExecutorManagerException
    */
   private void deletePod(final int executionId) throws ExecutorManagerException {
+    final ExecutableFlow flow = this.executorLoader.fetchExecutableFlow(executionId);
```

Comment:
Note that `deletePods` is expected to be a non-blocking call (or close to one). Adding multiple db queries, and k8s-api calls and  esp any sleep violates that assumption and can create problems for callers of this method as they will indirectly be blocked.
Any thoughts around restructuring things differently ? (happy to discuss offline)

Reply:
That is a wonderful point. The idea is that we must always delete pods, right? There can't be an exception thrown which would prevent pod deletion and make it so that pods accumulate in the k8s cluster?

Reply:
Yes, another point of the comment was that any delay added within `deletePod` method will automatically become a delay for the callers of deletePod. Some such callers, such as FlowStatusListenerManager are expected to be running mostly async tasks, possibly a very high number of them though and there processing could get delayed. 
We discussed this offline briefly, noting here for reference.